lib/PreferenceList.cc

```diff
@@ -90,10 +70,22 @@ std::ostream& operator<<(std::ostream& out, const PreferenceList* pl) {
     std::stringstream stmp;
 
     for (auto i = pl->cbegin(), e = pl->cend(); i != e; ++i) {
-        stmp << i->vertex->get_id();
-
+        if(i->vertex != nullptr){
+            stmp << i->vertex->get_id();
+        }
+        else{
+            int index = i-pl->cbegin();
```

Comment:
Use std::distance to find distance between two iterators, change type of index to auto. https://en.cppreference.com/w/cpp/iterator/distance